### PR TITLE
Build API documentation 

### PR DESCRIPTION
**Pull request by tinatn29**
built docs with `auto_api.py` per cookiecutting instructions (files in doc/source were modified) + fixed typos flagged by codespell. 

Besides fixing the typos, I didn't make any tweaks after `auto_api` did its thing, so let me know if any of the files modified should be different...

did this build one of the two docs (or both) that @bobleesj mentioned in #1218?

### CODE REVIEW

File: doc/source/collections/abstracts.rst
Comment (bobleesj) [inline doc/source/collections/abstracts.rst:1]: why is this file added?

File: doc/source/collections/projecta.rst [hunk 1]
```diff
@@ -45,7 +45,7 @@ The following lists key names mapped to its type and meaning for each entry.
 	:notes: list, any notes about the kickoff, optional
 	:status: string, status of the kickoff. Allowed values are {', '.join(PROJECTUM_STATI)}, optional
 
-		Allowed values:
```
Comment (bobleesj) [inline doc/source/collections/projecta.rst:48]: not sure the cause of these extra spaces provided throuhgout. I think this PR should only fix along `doc/source/api/...` right?

**Review by bobleesj — COMMENTED** [review]
I had a preliminary look.  Please check my in-line comments. @tinatn29 

For this PR, since we are modifying a lot, I think it would be safer for us to just do api_doc only? Also since there is one commit, I can't really track what manual modifications have been made.

**Comment by tinatn29** [comment]
will redo this PR to separate commits and make sure `auto_api` is doing the right thing
